src/transformStateWithClones.js

```diff
@@ -6,8 +6,34 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateHistory = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        currentState = { ...currentState };
```

Comment:
[CODE STYLE] - Creating a new object with `{ ...currentState }` is unnecessary here since you are already making a copy of `currentState` in the previous `case` blocks. This line can be removed to avoid redundant object copying.